elyra/metadata/metadata_app.py

```diff
@@ -766,6 +766,122 @@ def start(self):
+            try:
+                schema_name = metadata_file["schema_name"]
+                display_name = metadata_file["display_name"]
+                metadata = metadata_file["metadata"]
+            except KeyError as e:
+                non_imported_files.append([file, f"Could not find '{e.args[0]}' key in the import file '{filepath}'"])
+                continue
+
+            try:
+                if self.overwrite_flag.value:  # if overwrite flag is true
+                    try:  # try updating the existing instance
+                        updated_instance = self.metadata_manager.get(name)
+                        updated_instance.schema_name = schema_name
+                        if display_name:
+                            updated_instance.display_name = display_name
+                        if name:
+                            updated_instance.name = name
+                        updated_instance.metadata.update(metadata)
+                        self.metadata_manager.update(name, updated_instance)
+                    except MetadataNotFoundError:  # no existing instance - create new
+                        instance = Metadata(
+                            schema_name=schema_name, name=name, display_name=display_name, metadata=metadata
+                        )
+                        self.metadata_manager.create(name, instance)
+                else:
+                    instance = Metadata(
+                        schema_name=schema_name, name=name, display_name=display_name, metadata=metadata
+                    )
+                    self.metadata_manager.create(name, instance)
+            except Exception as e:
+                if isinstance(e, MetadataExistsError):
+                    non_imported_files.append([file, f"{str(e)} Use --overwrite to update."])
+                else:
+                    non_imported_files.append([file, str(e)])
+
+        instance_count_not_imported = len(non_imported_files)
+        instance_count_imported = len(json_files) - instance_count_not_imported
+
+        print(f"Imported {instance_count_imported} " + ("instance" if instance_count_imported == 1 else "instances"))
+
+        if instance_count_not_imported > 0:
+            print(
+                f"{instance_count_not_imported} "
+                + ("instance" if instance_count_not_imported == 1 else "instances")
+                + " could not be imported"
+            )
+
+            non_imported_files.sort(key=lambda x: x[0])
+            print("\nThe following files could not be imported: ")
+
+            # pad to width of longest file and reason
+            max_file_name_len = len("File")
+            max_reason_len = len("Reason")
+            for file in non_imported_files:
+                max_file_name_len = max(len(file[0]), max_file_name_len)
+                max_reason_len = max(len(file[1]), max_reason_len)
+
+            print("%s   %s" % ("File".ljust(max_file_name_len), "Reason".ljust(max_reason_len)))
```

Comment:
Can you use f-strings? https://docs.python.org/3/library/string.html#format-specification-mini-language

Reply:
@ptitzler Will do - should I also replace similar print statements used in SchemaspaceList ?